packages/attune/tests/smoke/main.rs

```diff
@@ -0,0 +1,465 @@
+    
+    // Create GPG key configuration file
+    let gpg_config = r#"Key-Type: EDDSA
+Key-Curve: Ed25519
+Subkey-Type: ECDH
+Subkey-Curve: Cv25519
+Name-Real: Attune Test
+Name-Email: test@attunehq.com
+Expire-Date: 0
+%no-protection
+%commit
+"#;
+    
+    let config_path = "/tmp/gpg_key_config.txt";
+    
+    match fs::write(config_path, gpg_config) {
+        Ok(_) => println!("✅ GPG config file created successfully"),
+        Err(e) => {
+            eprintln!("❌ Failed to create GPG config file: {e}");
+            panic!("GPG config file creation failed");
+        }
+    }
+    
+    // Generate GPG key using batch mode and capture output
+    println!("Generating GPG key (this may take a moment)...");
+    let gpg_cmd = cmd!(sh, "gpg --batch --generate-key {config_path}");
+    let gpg_generate_result = gpg_cmd.output();
+    
+    match gpg_generate_result {
+        Ok(output) => {
+            if !output.status.success() {
+                eprintln!("❌ GPG key generation failed with exit code: {}", output.status);
+                eprintln!("stdout: {}", String::from_utf8_lossy(&output.stdout));
+                eprintln!("stderr: {}", String::from_utf8_lossy(&output.stderr));
+                panic!("GPG key generation failed. Make sure GPG is installed and configured.");
+            }
+            
+            println!("✅ GPG key generated successfully");
+            let combined_output = format!("{}{}", 
+                String::from_utf8_lossy(&output.stdout),
+                String::from_utf8_lossy(&output.stderr)
+            );
+            println!("GPG output:\n{combined_output}");
+            
+            // Parse the GPG output to find the key ID from the revocation certificate message
+            for line in combined_output.lines() {
+                if line.contains("revocation certificate stored as") && line.contains(".rev'") {
+                    // Extract the key ID from the filename
+                    if let Some(start) = line.rfind('/') {
+                        if let Some(end) = line.rfind(".rev'") {
+                            let key_id = line[(start + 1)..end].to_string();
+                            println!("✅ Extracted GPG key ID from revocation certificate: {key_id}");
+                            
+                            // Clean up config file
+                            if let Err(e) = fs::remove_file(config_path) {
+                                eprintln!("⚠️  Warning: Could not clean up GPG config file: {e}");
+                            }
+                            
+                            // Set the GPG_KEY_ID environment variable
+                            sh.set_var("GPG_KEY_ID", &key_id);
+                            println!("✅ GPG_KEY_ID environment variable set");
+                            
+                            return key_id;
+                        }
+                    }
+                }
+            }
```

Comment:
[optional] consider using [`nom`](https://docs.rs/nom/latest/nom/) to parse this.
no expectation though- learning how to use parser combinators is definitely beyond what i'd expect a non-engineer skillset to be. so only if you want nerd points. this is fine for now.

Reply:
Makes sense. Added TODO comment.